ext/gvl_tracing_native_extension/gvl_tracing.c

```diff
@@ -163,29 +231,29 @@ static double timestamp_microseconds(void) {
   return (current_monotonic.tv_nsec / 1000.0) + (current_monotonic.tv_sec * 1000.0 * 1000.0);
 }
 
-static void set_native_thread_id(void) {
+static void set_native_thread_id(thread_local_state *state) {
   uint64_t native_thread_id = 0;
 
   #ifdef HAVE_PTHREAD_THREADID_NP
     pthread_threadid_np(pthread_self(), &native_thread_id);
   #elif HAVE_GETTID
     native_thread_id = gettid();
   #else
-    native_thread_id = current_thread_serial; // TODO: Better fallback for Windows?
+    native_thread_id = state->current_thread_serial; // TODO: Better fallback for Windows?
   #endif
 
-  thread_id = native_thread_id;
+  state->thread_id = native_thread_id;
 }
```

Comment:
I guess this may be completely incorrect for 3.3? E.g. the assumption for `gettid()` and `pthread_self()` is that they get called from the thread that the event is about, but if events are emitted from other threads now then this is wrong?

Reply:
Yeah, the alternative could be to call `Thread#native_thread_id`. But you need the GVL for that, so to me it's akin to `Thread#name`, some things to display, but not a good "primary key".